windowsforms/FlexGridShowcaseDemo/cs/MainForm.cs

```diff
@@ -0,0 +1,682 @@
+            countryColumn.ImageMap = flagImageMap;
+            countryColumn.ImageAndText = true;
+
+            // build image map for colors
+            var colorImageMap = new Dictionary<DrawColors, Image>();
+            foreach (DrawColors color in Enum.GetValues(typeof(DrawColors)))
+            {
+                colorImageMap.Add(color, LoadImage(color.ToString()));
+            }
+
+            var colorColumn = columns["Color"];
+
+            // assign image map to color column
+            colorColumn.ImageMap = colorImageMap;
+            colorColumn.ImageAndText = true;
+
+            var priceColumn = columns["Price"];
+            priceColumn.Format = "C2";
+            priceColumn.TextAlign = TextAlignEnum.RightCenter;
+            priceColumn.Width = 80;
+
+            var priceEditorValidation = _flexGrid.Cols["Price"].EditorValidation;
+
+            // add validation rules
+            priceEditorValidation.Add(new RequiredRule());
+            priceEditorValidation.Add(new RangeRule()
+            {
+                Minimum = decimal.Zero,
+                Maximum = decimal.MaxValue,
+                ErrorMessage = "Price cannot be negative"
+            });
+
+            var changeColumn = columns["Change"];
+            changeColumn.Format = "C2";
+            changeColumn.TextAlign = TextAlignEnum.RightCenter;
+
+            var historyColumn = columns["History"];
+            historyColumn.AllowEditing = false;
+
+            // setup sparkline for history column 
+            historyColumn.ShowSparkline = true;
+
+            var historySparkline = historyColumn.Sparkline;
+            historySparkline.ShowLow = true;
+            historySparkline.ShowHigh = true;
+
+            var discountColumn = columns["Discount"];
+            discountColumn.Format = "p0";
+            discountColumn.AllowEditing = false;
+            discountColumn.Width = 80;
+
+            var ratingColumn = columns["Rating"];
+            ratingColumn.ImageAndText = false;
+            ratingColumn.AllowEditing = false;
+
+            columns["Active"].Width = 60;
+
+            var dateColumn = columns["Date"];
+            dateColumn.Format = "g";
+
+            // creating footers
+            var footerDescription = new FooterDescription();
+
+            // price aggregate
+            var priceAggregateDefinition = new AggregateDefinition();
+            priceAggregateDefinition.Aggregate = AggregateEnum.Average;
+            priceAggregateDefinition.Caption = "Average price: ${0:N2}";
+            priceAggregateDefinition.PropertyName = "Price";
+
+            // discount aggregate
+            var discountAggregateDefinition = new AggregateDefinition();
+            discountAggregateDefinition.Aggregate = AggregateEnum.Average;
+            discountAggregateDefinition.Caption = "Average discount: {0:P}";
+            discountAggregateDefinition.PropertyName = "Discount";
+
+            var aggregates = footerDescription.Aggregates;
```

Comment:
I would use explicit typing in all places where the type is not apparent from the right side.

Reply:
I hope I fixed this.
Please, take a look - 2beb8aa68d9d689bc678a124446d26ca49b2f746.